cocos/network/CCWebSocket-android.cpp

```diff
@@ -0,0 +1,403 @@
+                                           "disconnect",
+                                           "(JZ)V")) {
+            jlong connectionID = cid;
+            methodInfo.env->CallStaticVoidMethod(methodInfo.classID, methodInfo.methodID,
+                                                 connectionID, (jboolean) syncClose);
+            methodInfo.env->DeleteLocalRef(methodInfo.classID);
+        }
+
+    }
+
+
+    void _callJavaSendBinary(int64_t cid, const unsigned char *data, size_t len) {
+
+        JniMethodInfo methodInfo;
+        if (JniHelper::getStaticMethodInfo(methodInfo,
+                                           J_BINARY_CLS_WEBSOCKET,
+                                           "sendBinary",
+                                           "(J[B)V")) {
+            jlong connectionID = cid;
+            jbyteArray jdata = methodInfo.env->NewByteArray((jsize) len);
+            methodInfo.env->SetByteArrayRegion(jdata, 0, (jsize) len, (const jbyte *) data);
+            methodInfo.env->CallStaticVoidMethod(methodInfo.classID, methodInfo.methodID,
+                                                 connectionID, jdata);
+            methodInfo.env->DeleteLocalRef(jdata);
+            methodInfo.env->DeleteLocalRef(methodInfo.classID);
+        }
+
+    }
+
+    void _callJavaSendString(int64_t cid, const std::string &data) {
+        JniMethodInfo methodInfo;
+        if (JniHelper::getStaticMethodInfo(methodInfo,
+                                           J_BINARY_CLS_WEBSOCKET,
+                                           "sendString",
+                                           "(J"  JARG_STR ")V")) {
+            jlong connectionID = cid;
+            jstring jdata = methodInfo.env->NewStringUTF(data.c_str());
+            methodInfo.env->CallStaticVoidMethod(methodInfo.classID, methodInfo.methodID,
+                                                 connectionID, jdata);
+            methodInfo.env->DeleteLocalRef(jdata);
+            methodInfo.env->DeleteLocalRef(methodInfo.classID);
+        }
+    }
+
+    static JNINativeMethod sMethodTable[] = {
+            {"triggerEvent", "(J" JARG_STR JARG_STR "Z)V", (void *) cocos2d::network::_WebSocketAndroidNativeTriggerEvent}
+    };
+
+    static bool _registerNativeMethods(JNIEnv *env) {
+        jclass clazz = env->FindClass(JCLS_WEBSOCKET);
+        if (clazz == nullptr) {
+            CCLOGERROR("_registerNativeMethods: can't find java class:%s", JARG_DOWNLOADER);
+            return false;
+        }
+        if (JNI_OK != env->RegisterNatives(clazz, sMethodTable,
+                                           sizeof(sMethodTable) / sizeof(sMethodTable[0]))) {
+            CCLOGERROR("_registerNativeMethods: failed");
+            if (env->ExceptionCheck()) {
+                env->ExceptionClear();
+            }
+            return false;
+        }
+        return true;
+    }
+
+    static bool __registered = false;
```

Comment:
1. Please avoid to use static variable
2. Variable can not start from __

17.4.3.1.2 Global names [lib.global.names]
Certain sets of names and function signatures are always reserved to the implementation:

Each name that contains a double underscore (__) or begins with an underscore followed by an uppercase letter (2.11) is reserved to the implementation for any use.
Each name that begins with an underscore is reserved to the implementation for use as a name in the global namespace.165
165) Such names are also reserved in namespace ::std (17.4.3.1).